### ISSUE TITLE
Add `safeAfter` helper function to `Maybe` name space

### ISSUE THREAD

**Issue by evilsoft** [evilsoft/crocks#207]
With all those crazy `value | undefined` or `Something | Another | Something Else` returning functions in the wild it may be handy to take care of lifting the result of functions into a `Maybe` based on a some predicate. While `safeLift` can be used to check an input before calling a function, the typical `compose(safe(pred), someFunc)` can be replaced with this `safeAfter` function. It has the following signature:

```hs
safeLift :: ((a -> Boolean) | Pred) -> (a -> b) -> a -> Maybe b
```

This will need the following to be accomplished:
* Add code:
  - [ ] `src/Maybe/safeAfter.js`
* Add specs
  - [ ] `src/Maybe/safeAfter.spec.js`
* Documentation
  - [ ] Update ReadMe (`src/Maybe/README.md`)
  - [ ] Add to Docs with description and example usage (`docs/src/pages/docs/crocks/Maybe.md`)
  - [ ] Add link to Helpers Section (`docs/src/pages/docs/functions/index.md`)
* Expose new functions to public API (`src/index.js`)
  - [ ] Entry for` safeAfter`
* Add specs for new public functions (`src/index.spec.js`)
  - [ ] Entry for` safeAfter`

**Comment by karthikiyengar** [comment]
Unless you're already working on this one, I'd be happy to take a stab.

**Comment by evilsoft** [comment]
Heck yesh. Sorry I did not add the Add `Setoid` to the  `Monoid`s. I will add that tomorrow. Got caught up on playing with the experimental es modules.

**Comment by karthikiyengar** [comment]
A question about the signature here. Shouldn't it be `safeAfter :: ((b -> Boolean) | Pred) -> (a -> b) -> Maybe b`? Am I missing something here?

Here's how I read it in my head - We execute a function from `a -> b`, pass `b` to a predicate, and we get a `Maybe b` depending on the predicate.

**Comment by evilsoft** [comment]
The only thing missing in your signature is the `a` we need to take in.
But we do not want to bring Pred in and should look something like:

```js
// safeAfter :: ((b -> Boolean) | Pred) -> (a -> b) -> a -> Maybe b
function safeAfter(pred, fn) {
  const predFn =
    isFunction(pred) ? pred : pred.runWith

  return x => {
    const result = fn(x)
    return predFn(result)
      ? Just(x)
      : Nothing()
  }
}
```

Using the `Pred` proxy to keep `Pred` out of core.

**Comment by evilsoft** [comment]
After playing with the two options for a bit, I say lets go with the first that only allow unary functions and let the user partially apply up to the last arg. It adds less complexity and gets it in the lib quicker to get usage feedback. If it turns out a lot of people would prefer the second option, then we can add the complexity at that time.
Thoughts?

**Comment by karthikiyengar** [comment]
Sure, I agree with the unary approach - keeps it predictable.

**Comment by karthikiyengar** [comment]
@evilsoft - Also, I'm pretty sure it's a typo, but I just wanted to confirm for good measure - the returned function should be returning `Just(result)` right (instead of `Just(x)`).

**Comment by evilsoft** [comment]
@karthikiyengar good catch...fixing

**Comment by evilsoft** [comment]
This has been completed and merged, will go out in the next release `0.9.1`
